debug/debug.go

```diff
@@ -361,22 +370,28 @@ func StopProfileBuffers(ctx context.Context) {
 		err = pent.WriteTo(v.buf, debug)
 		if err != nil {
 			log(ctx).With("cause", err).Warn("error writing PPROF buffer")
-
 			continue
 		}
 	}
 	// dump the profiles out into their respective PEMs
 	for k, v := range pprofConfigs.pcm {
+		// process context
+		if ctx.Err() != nil {
+			log(ctx).With("cause", ctx.Err()).Error("cannot write PEM")
+			return
+		}
+
```

Comment:
nit; What happens if we do not check the process ctx for an error in this loop ?